manifests/crc/BYODB-installation/openshift/kruize-crc-openshift.yaml

```diff
@@ -68,7 +68,22 @@ data:
           "secretAccessKey": "",
           "logLevel": "INFO"
         }
-      }
+      },
+      "datasource": [
+        {
+          "name": "prometheus-1",
+          "provider": "prometheus",
+          "serviceName": "prometheus-k8s",
+          "namespace": "openshift-monitoring",
+          "url": "",
+          "authentication": {
```

Comment:
Ensure that this change, including the check for the authentication object, is applied only when local=true. The Kruize pods should not fail if the authentication object is absent, as sometimes datasource authentication is not required.

Reply:
The datasource object is applicable only for local monitoring. Therefore, any code changes related to it have been made specifically within the local monitoring context and do not affect the ROS case.